sdks/go/pkg/beam/core/runtime/harness/worker_status_test.go

```diff
@@ -0,0 +1,83 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package harness
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"sync"
+	"testing"
+
+	fnpb "github.com/apache/beam/sdks/v2/go/pkg/beam/model/fnexecution_v1"
+	"google.golang.org/grpc"
+)
+
+type BeamFnWorkerStatusServicer struct {
+	fnpb.UnimplementedBeamFnWorkerStatusServer
+	response chan string
+}
+
+func (w BeamFnWorkerStatusServicer) mustEmbedUnimplementedBeamFnWorkerStatusServer() {
```

Comment:
Nit: You shouldn't need to implement this, since you've embedded it the servicer status.

You're running into that you're using a value, rather than a pointer method receiver for WorkerStatus. 

Change `(w BeamFnWorkerStatusServicer)` to `(w *BeamFnWorkerStatusServicer)`  and   `srv := BeamFnWorkerStatusServicer` to `srv := &BeamFnWorkerStatusServicer` and it should clear up.  (and save you from needing to do `&srv` later. Either will be stack or heap allocated as needed.)